app/src/main/java/com/example/desafios_zup/view/search/RecyclerViewSearchAdapter.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.desafios_zup.view.search
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Filter
+import android.widget.Filterable
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.desafios_zup.R
+import com.example.desafios_zup.view.MoviesModel
+import kotlinx.android.synthetic.main.movie_item.view.*
+import java.util.*
+import kotlin.collections.ArrayList
+
+class RecyclerViewSearchAdapter() :
+    RecyclerView.Adapter<RecyclerViewSearchAdapter.MovieHolder>(), Filterable {
+
+    var moviesFilterList: List<MoviesModel> = arrayListOf()
+
+    lateinit var mcontext: Context
+
+    class MovieHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val capa: ImageView = itemView.capa
+        val nome: TextView = itemView.nome
+        val genero: TextView = itemView.genero
+        val pais: TextView = itemView.pais
+        val sinopse: TextView = itemView.sinopse
+    }
+
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieHolder {
+        val moviesListView = LayoutInflater.from(parent.context)
+            .inflate(R.layout.movie_item, parent, false)
+        val sch = MovieHolder(moviesListView)
+        mcontext = parent.context
+        return sch
+    }
+
+    override fun getItemCount(): Int {
+        return moviesFilterList.size
+    }
+
+    override fun onBindViewHolder(holder: MovieHolder, position: Int) {
+        val currentItem = moviesFilterList[position]
+        holder.capa.setImageResource(currentItem.imageResource)
+        holder.nome.text = currentItem.nome
+        holder.genero.text = currentItem.genero
+        holder.pais.text = currentItem.pais
+        holder.sinopse.text = currentItem.sinopse
+    }
+
+    fun filterList(charFilter : String): List<MoviesModel> {
+        val resultList = ArrayList<MoviesModel>()
+        for (row in moviesFilterList) {
+            if (row.nome.toLowerCase(Locale.ROOT)
```

Comment:
o conteudo desse If ficou bem grande, você poderia transformar em um método boolean, e aqui só chama-lo 
```suggestion
            if (containsCharFilter(row.nome, charFilter))
```

Reply:
done